src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -1329,9 +1329,13 @@ class spell_gen_gift_of_naaru : public AuraScript
         if (!GetCaster())
             return;
 
-        float heal = 0.2f * GetCaster()->GetMaxHealth();
-        int32 healTick = std::floor(heal / aurEff->GetTotalTicks());
-        amount += int32(std::max(healTick, 0));
+        if (SpellEffectInfo const* eff1 = GetSpellInfo()->GetEffect(EFFECT_1))
+        {
+            float healPct = eff1->CalcValue() / 100.0f;
+            float heal = 0.2f * GetCaster()->GetMaxHealth();
```

Comment:
you forgot to replace 0.2f with healPct here :D

Reply:
lol just saw that too xd